autoload/ale/fixers/puppetlint.vim

```diff
@@ -0,0 +1,14 @@
+" Author: Alexander Olofsson <alexander.olofsson@liu.se>
+" Description: puppet-lint fixer
+
+function! ale#fixers#puppetlint#Fix(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'puppet_puppetlint_executable')
```

Comment:
You'll need to default initialise the variables above, in case someone hasn't loaded the linter, but has loaded the fixer.